core/src/impl/Kokkos_Core.cpp

```diff
@@ -592,21 +598,22 @@ void initialize(int& narg, char* arg[])
             auto env_skip_device = std::strtol(env_skip_device_str,&endptr,10);
             if (endptr== env_skip_device_str) 
                 Impl::throw_runtime_exception("Error: cannot convert KOKKOS_SKIP_DEVICE to an integer. Raised by Kokkos::initialize(int narg, char* argc[]).");
-            if (errno == ERANGE && (env_skip_device == LONG_MAX || env_skip_device == LONG_MIN))
+            if (errno == ERANGE)
                 Impl::throw_runtime_exception("Error: KOKKOS_SKIP_DEVICE out of range of representable values by an integer. Raised by Kokkos::initialize(int narg, char* argc[]).");
             if ((skip_device != 9999)&&(env_skip_device!=skip_device))
-                Impl::throw_runtime_exception("Error: expecting a match between --kokkos-ndevices and KOKKOS_SKIP_DEVICE if both are set'. Raised by Kokkos::initialize(int narg, char* argc[]).");
+                Impl::throw_runtime_exception("Error: expecting a match between --kokkos-ndevices and KOKKOS_SKIP_DEVICE if both are set. Raised by Kokkos::initialize(int narg, char* argc[]).");
             else
                 skip_device = env_skip_device;
         }
     }
-    const char * env_disablewarnings_str = std::getenv("KOKKOS_DISABLE_WARNINGS");
+    char * env_disablewarnings_str = std::getenv("KOKKOS_DISABLE_WARNINGS");
     if (env_disablewarnings_str!=nullptr) {
-        if (strcmp(env_disablewarnings_str,"TRUE") == 0)
+        Impl::convert_uppercase(env_disablewarnings_str);
```

Comment:
["Modifying the string returned by getenv invokes undefined behavior."](https://en.cppreference.com/w/cpp/utility/program/getenv)  This is easy to fix: 
```c++
std::string env_str (env_disablewarnings_str); // deep-copies input
for (char& c : env_str) { c = toupper (c); }
```

Reply:
@mhoemmen I did not know that. Thanks for pointing it out. Code was fixed.